Subject: [PATCH] Fix PacketKeepalive.stop() crash

If the keepalive stop() function is called when the mSlot number is
not yet setup, the conversion from null to integer will crash.
This patch fixes it by adding the error checking.

diff --git a/core/java/android/net/ConnectivityManager.java b/core/java/android/net/ConnectivityManager.java
index a45e6f5..36d86ed 100644
--- a/core/java/android/net/ConnectivityManager.java
+++ b/core/java/android/net/ConnectivityManager.java
@@ -1529,6 +1529,9 @@
         }
 
         public void stop() {
+            if (mSlot == null) {
+                return;
+            }
             try {
                 mService.stopKeepalive(mNetwork, mSlot);
             } catch (RemoteException e) {
